File: app/src/main/java/net/sytes/kai_soft/letsbuyka/ProductModel/DetailFragment.java
package net.sytes.kai_soft.letsbuyka.ProductModel;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;

import net.sytes.kai_soft.letsbuyka.CRUDdb;
import net.sytes.kai_soft.letsbuyka.IMenuContract;
import net.sytes.kai_soft.letsbuyka.R;

/**
 * Created by Лунтя on 30.04.2018.
 */

public class DetailFragment extends Fragment implements IMenuContract {

    //Button insertBtn, backToListBtn, deleteBtn;
    EditText etName, etDescr;//, etPhoto;

    IProductListActivityContract IProductListActivityContract;



    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_detail, container, false);

        //insertBtn = rootView.findViewById(R.id.insertBtn);
        //backToListBtn = rootView.findViewById(R.id.backToListBtn);
        //deleteBtn = rootView.findViewById(R.id.deleteBtn);

        etName = rootView.findViewById(R.id.etName);
        etDescr = rootView.findViewById(R.id.etDescr);
//        etPhoto = rootView.findViewById(R.id.etPhoto);




        //dbProduct = Application.getDB(); //new DataBase(this.getContext());

        return rootView;
    }

    @Override
    public void onResume() {
        super.onResume();

        makeEditable(true);

        emptyEditText();

        //backToListBtn.setOnClickListener(this);
        //insertBtn.setOnClickListener(this);
        //deleteBtn.setVisibility(View.GONE);

        Bundle bundle = getArguments();

        if (bundle.getBoolean("editable")) {
            //insertBtn.setText(R.string.save);

        } else {
            Product product = (Product) bundle.getSerializable("product");

            etName.setText(product.getName());
            etDescr.setText(product.getDescription());
  //          etPhoto.setText(product.getFirstImagePath());

            //makeEditable(false);

            //insertBtn.setText(R.string.edit);

            //deleteBtn.setOnClickListener(this);
            //deleteBtn.setVisibility(View.VISIBLE);
        }
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof IProductListActivityContract) {
            IProductListActivityContract = (IProductListActivityContract) context;
        }
    }

    @Override
    public void onDetach() {
        IProductListActivityContract = null;
        super.onDetach();
    }

    private boolean isEditable() {
        return etName.isEnabled();
    }

    private boolean isNewProduct() {
        Bundle bundle = getArguments();
        if (bundle.getBoolean("editable") == true) {
            return true;
        }
        return false;
    }

    private void makeEditable(boolean state) {
        etName.setEnabled(state);
        etDescr.setEnabled(state);
//        etPhoto.setEnabled(state);
    }

    private void emptyEditText() {
        etName.setText("");
        etDescr.setText("");
//        etPhoto.setText("");
    }

    private Product toUpdate(){
        Bundle bundle = getArguments();
        Product fromBundle = (Product) bundle.getSerializable("product");
        Product toUpdate = new Product(fromBundle.getId(),
                etName.getText().toString(),
                etDescr.getText().toString());
        return toUpdate;
    }


    @Override
    public void savePressed() {

        if (isNewProduct() == false) {
            Product updateble = toUpdate();
            CRUDdb.Companion.updateTableProducts(updateble);
            //IProductListActivityContract.onDetailFragmentButtonClick();
        } else {
            CRUDdb.Companion.insertToTableProducts(etName.getText().toString(),
                    etDescr.getText().toString());
            //IProductListActivityContract.onDetailFragmentButtonClick();
        }
    }

    @Override
    public void deletePressed(){
        Bundle bundle = getArguments();
        if (isNewProduct() == false) {
            CRUDdb.Companion.deleteItemProducts((Product) bundle.getSerializable("product"));
            //IProductListActivityContract.onDetailFragmentButtonClick();
        }
    }

    public boolean isNew(){
        return isNewProduct();
    }
}

File: app/src/main/java/net/sytes/kai_soft/letsbuyka/ProductModel/DetailFragment.kt
package net.sytes.kai_soft.letsbuyka.ProductModel

import android.annotation.SuppressLint
import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.inputmethod.InputMethodManager
import android.widget.EditText
import net.sytes.kai_soft.letsbuyka.*

import net.sytes.kai_soft.letsbuyka.Lists.List

class DetailFragment: Fragment(), IMenuContract {


    companion object {
        private const val myTag = "letsbuy_PRdetFragment"
    }

    private lateinit var etName: EditText
    private lateinit var etDesct: EditText
    private lateinit var list: List
    private lateinit var iActivityContract: IActivityContract
    private lateinit var iPRContract: IPRContract

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        Log.i(myTag, "onCreateView()")
        val rootView = inflater!!.inflate(R.layout.fragment_detail, container, false)
        etName = rootView.findViewById(R.id.etName)
        etDesct = rootView.findViewById(R.id.etDescr)
        return rootView
    }

    override fun onResume() {
        Log.i(myTag, "onResume()")
        super.onResume()

        emptyEditText()

        val bundle = arguments
        if (!bundle.getBoolean("editable")){
            val product = bundle.getSerializable(Constants.PRODUCT) as Product

            etName.setText(product.name)
            etDesct.setText(product.description)
        }
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        if (context is IActivityContract){
            iActivityContract = context
        }
        if (context is IPRContract){
            iPRContract = context
        }
    }

    private fun emptyEditText(){
        etName.text.clear()
        etDesct.text.clear()
    }

    fun isNewElement(): Boolean{
        val bundle = arguments
        return bundle.getBoolean(Constants.EDITABLE)
    }

    private fun toUpdate(): Product {
        val bundle = arguments
        val fromBundle = bundle.getSerializable(Constants.PRODUCT) as Product
        return Product(fromBundle.id, etName.text.toString(), etDesct.text.toString())
    }

    override fun savePressed() {
        if (!isNewElement()){
            val updatable = toUpdate()
            CRUDdb.updateTableProducts(updatable)
            iActivityContract.onDetailFragmentButtonClick()
        } else {
            CRUDdb.insertToTableProducts(etName.text.toString(),
                    etDesct.text.toString())
            iActivityContract.onDetailFragmentButtonClick()
        }
    }

    override fun deletePressed() {
        val bundle = arguments
        CRUDdb.deleteItemProducts(bundle.getSerializable(Constants.PRODUCT) as Product)
        iActivityContract.onDetailFragmentButtonClick()
    }

    override fun hideKeyboard() {

        Log.i(myTag, "hideKeyboard()")
        try{
            val imm = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
            imm.hideSoftInputFromWindow(etName.windowToken, 0)
        }
        catch (e: NullPointerException){}

    }

}